src/runtime/js/ts.ui/core/core-gui@tradeshift.com/spirits/modals/ts.ui.ModalSpirit.js

```diff
@@ -72,23 +72,8 @@ ts.ui.ModalSpirit = (function using(Client, transition, chained) {
 		 */
 		open: function(opt_open) {
 			var then = (this._then = new gui.Then());
-			opt_open = arguments.length ? opt_open : true;
-			this.css
-				.shift(this.$fullscreen, 'ts-fullscreen')
-				.shift(!this.$fullscreen, 'ts-inscreen')
-				.shift(this.$fullscreen, 'ts-overflow');
-			if (opt_open !== this.isOpen) {
-				if (opt_open) {
-					if (this._execute('onopen') && this._confirmposition()) {
-						this.isOpen = true;
-						this._fadeIn();
-					}
-				} else {
-					if (this._execute('onclose')) {
-						this.isOpen = false;
-						this._fadeOut();
-					}
-				}
+			if (!this.doorman.open(opt_open)) {
+				then.now(false); // TODO: such a good idea to return a "promise", then?
```

Comment:
The Aside and SideBar should, I guess, also be refactored to return promises (or at least fake "promises") when the animation is done. They currently return themselves (chained), so that might however break something.